scripts/replace_parameter.sh

```diff
@@ -1,27 +1,55 @@
 #!/bin/bash
 
 usage() {
-	echo './replace_parameter.sh <REGISTER> <VALUE> <LINK>'
+
+	echo './replace_parameter.sh [-f] <REGISTER> <LINK> <VALUE/FILENAME>'
 	echo ''
 	echo '	REGISTER - register to be updated dropping the "CFG_" substring'
 	echo ''
-	echo '	VALUE - value in decimal to be assigned to REGISTER'
+	echo '	VALUE/FILENAME - value in decimal to be assigned to REGISTER or, if the -f option is used, filename containing a list of vfat number/ value pairs'
 	echo ''
-	echo '	Example:'
+	echo '	Examples:'
 	echo ''
 	echo '		./replace_parameter.sh PULSE_STRETCH 4 1'
-	echo ''
+	echo '		./replace_parameter.sh -f PULSE_STRETCH 4 /path/to/NominalDacValues.txt'
+	echo ''    
 
 	kill -INT $$; 
 }
 
-if [ -z ${3+x} ]
-then
-	usage
-fi
+ISFILE=0;
+OPTIND=1
+while getopts "fwhd" opts
+do
+    case $opts in
+        f)
+            ISFILE=1;;
+        h)
+            usage;;
+        \?)
+            usage;;
+        [?])
+            usage;;
+    esac
+done
+unset OPTIND
 
-REGISTER=$1
-VALUE=$2
-LINK=$3
-
-sed -i "s|.*${REGISTER}.*|${REGISTER}	${VALUE}|g" /mnt/persistent/gemdaq/vfat3/config_OH${LINK}_VFAT*_cal.txt
+if (($ISFILE == 1)); then
+    if [ -z ${4+x} ]
+    then
+	    usage
```

Comment:
Probably some statement here is needed about *which* usage example is expected.  You're using the same help menu for each output.  I think an `echo` after the call of usage to specify the file input case is sufficient.

Same comment for the below.

Reply:
done